api/core/routes.py

```diff
@@ -302,42 +255,195 @@ def get_ephemeris_by_catalog_number_jdstep():
+    stopjd: 'float'
+        UT1 Universal Time Julian Date to stop ephmeris calculation.
+    stepjd: 'float'
+        UT1 Universal Time Julian Date timestep.
+
+    Returns
+    -------
+    response: 'dictionary'
+        JSON output with satellite information - see jsonOutput() for format
+    '''
+
+    
+    tle = request.args.get('tle')
+    latitude = request.args.get('latitude')
+    longitude = request.args.get('longitude')
+    elevation = request.args.get('elevation')
+    startjd = request.args.get('startjd')
+    stopjd = request.args.get('stopjd')
+    stepjd = request.args.get('stepjd')
+    
+
+    #check for mandatory parameters
+    if [x for x in (latitude, longitude, elevation, startjd, stopjd, stepjd) if x is None]:
+        abort(400) 
+
+    #Cast the latitude, longitude, and jd to floats (request parses as a string)
+    try:
+        location = EarthLocation(lat=float(latitude)*u.deg, lon=float(longitude)*u.deg, height=float(elevation)*u.m)
+    except:
+        abort(500, "Error: Invalid location parameters")
+    
+    jd0 = float(startjd)
+    jd1 = float(stopjd) 
+    jds = float(stepjd)
+  
+    jd = jd_arange(jd0,jd1,jds)
+
+    if(len(jd)>1000):
+        app.logger.info("Too many results requested")
+        abort(400)
+
+    tleLine1, tleLine2, date_collected, name = parse_tle(tle)
+    return create_result_list(location, jd, tleLine1, tleLine2, date_collected, name)
+
 
 ### HELPER FUNCTIONS NOT EXPOSED TO API ###
 
-def create_result_list(lat, lon, ele, jd, identifier, use_catalog_number):
+def get_TLE(identifier, use_catalog_number):
     recent_TLE_sup, recent_TLE_gp = get_TLE_by_catalog_number(identifier) if use_catalog_number else get_TLE_by_name(identifier)
     tleLine1, tleLine2, date_collected, name = get_recent_TLE(recent_TLE_sup, recent_TLE_gp)
+    return tleLine1, tleLine2, date_collected, name
 
+
+def create_result_list(location, jd, tleLine1, tleLine2, date_collected, name):
     # propagation and create output
     resultList = []
     for d in jd:
         #Right ascension RA (deg), Declination Dec (deg), dRA/dt*cos(Dec) (deg/day), dDec/dt (deg/day),
         # Altitude (deg), Azimuth (deg), dAlt/dt (deg/day), dAz/dt (deg/day), distance (km), range rate (km/s), phaseangle(deg), illuminated (T/F)   
         [ra, dec, dracosdec, ddec, alt, az,  
-         r, dr, phaseangle, illuminated] = propagateSatellite(tleLine1,tleLine2,lat,lon,ele,d)
+         r, dr, phaseangle, illuminated] = propagateSatellite(tleLine1,tleLine2,location,d)
         
-        resultList.append(jsonOutput(name, d, ra, dec, date_collected, 
+        resultList.append(jsonOutput(name, d.jd, ra, dec, date_collected, 
                                     dracosdec, ddec,
                                     alt, az, 
                                     r, dr, phaseangle, illuminated)) 
     return resultList
 
+def parse_tle(tle):
+    #parse url encoded parameter to string to remove space encoding
```

Comment:
Please add full docstring for this and other functions, even if they are small and straightforward (bonus for type checking via "type hints" with `->`)